src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptions.cs

```diff
@@ -1,23 +1,60 @@
 ﻿namespace Microsoft.Extensions.DependencyInjection
 {
     using System.Collections.Generic;
+    using System.Linq;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.Extensions.Options;
 
     /// <summary>
-    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that uses <see cref="TelemetryConfiguration.Active"/> as initial value.
+    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that create new <see cref="TelemetryConfiguration"/> every time when called"/>.
     /// </summary>
     internal class TelemetryConfigurationOptions : IOptions<TelemetryConfiguration>
     {
         public TelemetryConfigurationOptions(IEnumerable<IConfigureOptions<TelemetryConfiguration>> configureOptions)
         {
-            foreach (var c in configureOptions)
+            // workaround for Microsoft/ApplicationInsights-dotnet#613
+            this.Value = TelemetryConfiguration.CreateDefault();
+
+            var configureOptionsArray = configureOptions.ToArray();
+            foreach (var c in configureOptionsArray)
             {
                 c.Configure(this.Value);
             }
+
+            // as we expect some customers to use TelemetryConfiguration.Active even together with DependencyInjection
+            // we make sure it has been set up
+            // it must be done only once even if there are multiple Web Hosts in the process
+            if (!IsActiveConfigured(this.Value.InstrumentationKey))
+            { 
+                foreach (var c in configureOptionsArray)
+                {
+                    c.Configure(TelemetryConfiguration.Active);
+                }
+            }
         }
 
         /// <inheritdoc />
-        public TelemetryConfiguration Value => TelemetryConfiguration.Active;
+        public TelemetryConfiguration Value { get; }
+
+        /// <summary>
+        /// Determines is TelemetryConfiguration.Active needs to be configured
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key configuration</param>
+        /// <returns>True is Active TelemertryConfiguration was previously configured</returns>
+        private bool IsActiveConfigured(string instrumentationKey)
+        {
+            var active = TelemetryConfiguration.Active;
+            if (string.IsNullOrEmpty(active.InstrumentationKey) && !string.IsNullOrEmpty(instrumentationKey))
+            {
+                return false;
+            }
+
+            if (active.TelemetryInitializers.Count <= 1 && active.TelemetryProcessors.Count <= 1)
```

Comment:
I just want to check: Does TelemetryProcessors.Count include the always-present final processor that sends the telemetry item to the channel? If so, cool. Otherwise, surely it's possible to have an active telemetry configuration with no initializers and no processors. It might be unusual for most ASP.NET Core apps, but still possible.

Reply:
Yes, this default processor is set up when the TelemetryConfiguration.Active is constructed.
Perhaps checking for processors count does not make sense. Likely it is still one even after configuration.

This code attempts to configure Active static singleton in the same way as it configured TelemetryConfiguration created for DI. The latter could not possibly have 1 TelemetryIntializer (at least at this point, initializers could be removed later, but now they have been just configured) - this method is called after all telemetry initializers are added and there are like 20 of them.

Note that also we don't really support the case of having multiple different configurations in the same process. And this PR does not really help with it. It just does some damage control for the co-existing TelemetryConfiguration.Active and TelemetryConfiguration DI singleton.

If a user manually modifies Active and removes everything AND runs multiple webhosts, then he would need to update Active in every WebHost he creates. That sounds like a weird scenario, though.